actions/actions.py

```diff
@@ -0,0 +1,44 @@
+#!/usr/bin/env python3
+# -*- coding: utf8 -*-
+from enum import Enum, unique
+import logging
+
+
+@unique
+class NfcEvent(Enum):
+    DETECT = 1
+    REMOVE = 2
+    REDETECT = 3
+
+
+def resolve(config: dict, event: NfcEvent, tag_id: str):
+    if tag_id not in config:
+        logging.warning("No mapping for tag " + tag_id)
+        return []
+
+    # logging.debug("Action " + event.name + " for tag " + tag_id)
+    config_tag = config[tag_id]
+
+    event_to_key_map = {
+        NfcEvent.DETECT: "ondetect" if "ondetect" in config_tag else "url",
+        NfcEvent.REMOVE: "onremove",
+        NfcEvent.REDETECT: "onredetect" if "onredetect" in config_tag else "ondetect" if "ondetect" in config_tag else "url"
+    }
+
+    event_key = event_to_key_map[event]
+
+    if event_key not in config_tag:
+        logging.debug("No event key '" + event_key + "' for tag " + tag_id)
+        return []
+
+    logging.info("Executing '" + config_tag['name'] + "'[" + event_key + "].")
+
+    # configure list of actions
+    if type(config_tag[event_key]) is list:
+        return config_tag[event_key]
+    # legacy: just one action
+    elif type(config_tag[event_key]) is dict:
```

Comment:
thanks for compatibility with old configs ;-)

Reply:
Sure thing. But at some point we will have to drop old versions of the config. 

One item on the roadmap is "config validation on startup". First step would be to just log a warning and still support older versions. Later validation of the latest config version would be enforced and startup would fail.